### PR TITLE
`change-query` cleanup

### PR DESCRIPTION
**Pull request by ranquild**
Changes:
- add or remove clauses
- columns now have ids and types, we use id for clauses
- clauses are referenced by indexes and not names
- errors are returned in json so that we handle columns with spaces or commas etc
- temporary disable viz tools (added `json5` suffix to ignore them for now). I'm going to re-enable them one by one in next PRs.

Demo https://www.loom.com/share/f15f6d00b22344bcb1c51d6c02b6aa44

How to verify:
- Open Orders table
- `created in the last 2 years`
- `sum by total`
- `group by category`
- `order by category`
- `limit to 3`
- `remove limit`
- `remove sorting`
- `remove grouping`
- `remove aggregation`
- `remove filters`

### CODE REVIEW

File: enterprise/backend/src/metabase_enterprise/metabot_v3/tools/change_visualization_settings.clj [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  (:require
+    [clojure.string :as str]
+    [medley.core :as m]
+    [metabase-enterprise.metabot-v3.tools.interface :as metabot-v3.tools.interface]
+    [metabase.util.log :as log]
+    [metabase.util.malli :as mu])
+  (:import
+   (clojure.lang ExceptionInfo)))
+
+(defn- column-error
```
Comment (ranquild) [inline enterprise/backend/src/metabase_enterprise/metabot_v3/tools/change_visualization_settings.clj:11]: Feedback loop

File: enterprise/frontend/src/metabase-enterprise/metabot/reactions/visualizations.ts [hunk 2]
```diff
@@ -49,26 +45,13 @@ export const changeDisplayType: ReactionHandler<
   reaction =>
   async ({ dispatch, getState }) => {
     {
-      const display = reaction.display_type;
       const question = getQuestion(getState());
-
       if (!question) {
-        dispatch(
```
Comment (ranquild) [inline enterprise/frontend/src/metabase-enterprise/metabot/reactions/visualizations.ts:56]: Handled by the tool

File: enterprise/backend/src/metabase_enterprise/metabot_v3/client.clj [hunk 3]
```diff
@@ -54,7 +54,7 @@
    session-id :- uuid?]
   (encode-request-body
    {:messages      messages
-    :context       {}
+    :context       (metabot-v3.context/hydrate-context (or context {}))
```
Comment (ranquild) [inline enterprise/backend/src/metabase_enterprise/metabot_v3/client.clj:57]: Rolls back how it was before

**Review by alxnddr — APPROVED** [review]
Looks good, worked well on many cases like "how many orders with total over 100 bucks were in this month?", "what's the most ordered product category?".

A few places to improve further I found:

**Orders without any discount**
Not aware it is nullable and applied filter discount = zero

**Only this and previous months**
Applied a relative date time filter "past 2 months" instead of past 1 month + "Include this month"

**Orders in jan 2024**

Added unnecessary current month filter 

<video src="https://github.com/user-attachments/assets/dfe0ad38-3f54-453f-a778-660867972859" />

**Pull request merged**